### ISSUE TITLE
DropOut

### ISSUE THREAD

**Issue by neworderofjamie** [genn-team/ml_genn#158]
DropOut is commonly used when training SNN models e.g. from the code accompanying http://arxiv.org/abs/2506.06374:

```python
def __init__(...):
    ...
    self.drop = nn.Dropout(p=dropout)
...
def forward(self, x):
    ...
    # Compute spikes via neuron dynamics
    s = self._lif_cell(Wx)

    # Apply dropout
    s = self.drop(s)

   return s
```

This would be easy to add to the eventprop compiler, it just requires ANDing an RNG call to the threshold condition. However, PyTorch also scales the spikes after dropout:

![Image](https://github.com/user-attachments/assets/089a13e8-8077-4090-b2b3-3a4999cafb8e)

But this scale factor could easily be applied postsynaptically
